extensions/azurecore/src/azureResource/tree/flatAccountTreeNode.ts

```diff
@@ -162,12 +107,126 @@ export class FlatAccountTreeNode extends AzureResourceContainerTreeNodeBase {
+	public readonly onNewResourcesAvailable = this._onNewResourcesAvailable.event;
+	private readonly _onLoadingStatusChanged = new vscode.EventEmitter<void>();
+	public readonly onLoadingStatusChanged = this._onLoadingStatusChanged.event;
+
+	constructor(private readonly appContext: AppContext,
+		private readonly _resourceService: AzureResourceService,
+		private readonly _subscriptionService: IAzureResourceSubscriptionService,
+		private readonly _subscriptionFilterService: IAzureResourceSubscriptionFilterService,
+		private readonly _account: AzureAccount,
+		private readonly _accountNode: TreeNode) {
+	}
+
+	public get isLoading(): boolean {
+		return this._isLoading;
+	}
+
+	public get nodes(): TreeNode[] {
+		return this._nodes;
+	}
+
+	public async start(): Promise<void> {
+		if (this._isLoading) {
+			return;
+		}
+		this._isLoading = true;
+		this._nodes = [];
+		this._onLoadingStatusChanged.fire();
+		let newNodesAvailable = false;
+
+		// Throttle the refresh events to at most once per 500ms
+		const refreshHandle = setInterval(() => {
+			if (newNodesAvailable) {
+				this._onNewResourcesAvailable.fire();
+				newNodesAvailable = false;
+			}
+			if (!this.isLoading) {
+				clearInterval(refreshHandle);
+			}
+		}, 500);
+		try {
+			let subscriptions: azureResource.AzureResourceSubscription[] = (await getSubscriptionInfo(this._account, this._subscriptionService, this._subscriptionFilterService)).subscriptions;
+
+			if (subscriptions.length !== 0) {
+				// Filter out everything that we can't authenticate to.
+				subscriptions = subscriptions.filter(async s => {
+					const token = await azdata.accounts.getAccountSecurityToken(this._account, s.tenant, azdata.AzureResource.ResourceManagement);
+					if (!token) {
+						console.info(`Account does not have permissions to view subscription ${JSON.stringify(s)}.`);
+						return false;
+					}
+					return true;
+				});
+			}
 
-	private _id: string = undefined;
-	private _label: string = undefined;
+			const resourceProviderIds = await this._resourceService.listResourceProviderIds();
+			for (const subscription of subscriptions) {
+				for (const providerId of resourceProviderIds) {
+					const resourceTypes = await this._resourceService.getRootChildren(providerId, this._account, subscription, subscription.tenant);
+					for (const resourceType of resourceTypes) {
+						const resources = await this._resourceService.getChildren(providerId, resourceType.resourceNode, true);
+						if (resources.length > 0) {
+							this._nodes.push(...resources.map(dr => new AzureResourceResourceTreeNode(dr, this._accountNode, this.appContext)));
+							this._nodes = this.nodes.sort((a, b) => {
+								return a.getNodeInfo().label.localeCompare(b.getNodeInfo().label);
+							});
+							newNodesAvailable = true;
+						}
+					}
+				}
+			}
+		} catch (error) {
+			if (error instanceof AzureResourceCredentialError) {
+				vscode.commands.executeCommand('azure.resource.signin');
```

Comment:
Is signing in going to restart the loading process then?

Reply:
nope, the loading process only starts when user tries to expand the node.